lib/rubocop/cop/gusto/ignored_columns_assignment.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Gusto
+      # Enforces proper usage of `ignored_columns` assignment
+      #
+      # This cop ensures that `ignored_columns` is assigned using `+=` with an array
+      # instead of direct assignment, which can cause issues with existing ignored columns.
+      #
+      # @example
+      #
+      #   # bad
+      #   self.ignored_columns = :column_name
+      #   self.ignored_columns = 'column_name'
+      #   self.ignored_columns = [:column_name]
+      #   self.ignored_columns = ['column_name']
+      #
+      #   # good
+      #   self.ignored_columns += [:column_name]
+      #   self.ignored_columns += ['column_name']
+      #
+      class IgnoredColumnsAssignment < Base
+        MSG = 'Use `+=` with an array for `ignored_columns` assignment instead of direct assignment.'
+        RESTRICT_ON_SEND = %i(ignored_columns=).freeze
+
+        # @!method ignored_columns_direct_assignment?(node)
+        def_node_matcher :ignored_columns_direct_assignment?, <<~PATTERN
+          (send (self) :ignored_columns= _)
+        PATTERN
+
+        def on_send(node)
+          if ignored_columns_direct_assignment?(node)
+            add_offense(node.loc.selector)
+          end
+        end
+        alias_method :on_csend, :on_send
```

Comment:
nit: we shouldn't need this and there's an exemption list for all class methods in the default.yml

but you can keep it in case someone was metaprogramming like `User.ignored_columns = ` which is of course very weird

Reply:
Oh I didn't think we should need it either but our own cops on this repo caught it. Is there something I can do to avoid that cop, or do I just make it a rubocop todo? @corsonknowles

Reply:
I would guess upstream will object to it, so we probably shouldn't keep

Reply:
oh okay I figured it out. I didn't fully understand what you meant about the exemption list at first.